app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -0,0 +1,342 @@
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.defaultMinSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Favicon
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SecondaryText
+import org.mozilla.fenix.theme.FirefoxTheme
+
+val LIST_ITEM_HEIGHT = 56.dp
+
+val ICON_SIZE = 24.dp
+
+/**
+ * List item Composable used to display text with an optional second line of text and
+ * an optional interactable icon at the end.
+ *
+ * @param label The primary text in the item.
+ * @param description An optional second line of text.
+ * @param onClick Optional lambda for handling clicks.
+ * @param icon Optional DrawableRes used to display a drawable at the end of the list item.
+ * @param iconDescription Content description of the icon.
+ * @param onIconClick Optional lambda for handling clicks on the icon.
+ */
+@Composable
+fun TextListItem(
+    label: String,
+    description: String? = null,
+    onClick: (() -> Unit)? = null,
+    @DrawableRes icon: Int? = null,
+    iconDescription: String? = null,
+    onIconClick: (() -> Unit)? = null,
+) {
+    ListItem(
+        label = label,
+        description = description,
+        onClick = onClick,
+        afterListAction = {
+            icon?.let {
+                ListItemIcon(
+                    icon = icon,
+                    iconDescription = iconDescription,
+                    onClick = onIconClick,
+                )
+
+                Spacer(modifier = Modifier.width(16.dp))
+            }
+        }
+    )
+}
+
+/**
+ * List item Composable used to display text and a [Favicon] with an optional second line of text and
+ * an optional interactable icon at the end.
+ *
+ * @param label The primary text in the item.
+ * @param description An optional second line of text.
+ * @param onClick Optional lambda for handling clicks.
+ * @param url The tab's URL.
+ * @param icon Optional DrawableRes used to display a drawable at the end of the list item.
+ * @param iconDescription Content description of the icon.
+ * @param onIconClick Optional lambda for handling clicks on the icon.
+ */
+@Composable
+fun FavIconListItem(
```

Comment:
```suggestion
fun FaviconListItem(
```
This happens to be a nice example that follows nicely with the Figma component naming. It would be nice if we can align our naming nicely with the Component System. I don't think we need a unique component for Left Favicon + Right Icon for instance, but we can take a more generic naming like Favicon List Item as the core component and know that the Right Icon, Descriptions and such will be optional way to extend a core component.